File: app/src/main/java/com/biryanistudio/todo/services/TodoStartCopyListenerServiceReceiver.java
package com.biryanistudio.todo.services;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * Created by aakaashjois on 12/05/17.
 */

public class TodoStartCopyListenerServiceReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction().equalsIgnoreCase(Intent.ACTION_BOOT_COMPLETED) ||
                intent.getAction().equalsIgnoreCase(Intent.ACTION_PACKAGE_REPLACED))
            context.startService(new Intent(context, CopyListenerService.class));
    }
}

File: app/src/main/java/com/biryanistudio/todo/services/TodoStartCopyListenerServiceReceiver.kt
package com.biryanistudio.todo.services

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * Created by Aakaash Jois.
 * 12/05/17 - 9:19 AM.
 */

class TodoStartCopyListenerServiceReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action.equals(Intent.ACTION_BOOT_COMPLETED, ignoreCase = true) ||
                intent.action.equals(Intent.ACTION_PACKAGE_REPLACED, ignoreCase = true))
            context.startService(Intent(context, CopyListenerService::class.java))
    }
}
